### PR TITLE
update: octoxlabs integration

### PR DESCRIPTION
**Pull request by ogulcanhero**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Octox Labs Cyber Security Asset Management platform XSoar Integration update.

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [x] 6.1.0
- [x] 6.2.0
- [x] 6.5.0
- [x] 6.12.0 

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml [hunk 1]
```diff
@@ -584,12 +595,237 @@ script:
...
+        name: scroll_id
+    outputs:
+      - contextPath: OctoxLabs.ScrolledApplications.count
+        description: 'Total applications count.'
+        type: Number
+      - contextPath: OctoxLabs.ScrolledApplications.scroll_id
+        description: 'Specific Scroll Id'
+        type: String
+      - contextPath: OctoxLabs.ScrolledApplications.results
+        description: 'List<Dict> Application information.'
```
Comment (MLainer1) [inline Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml:794]:
```suggestion
        description: 'List Application information.'
```

File: Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.py [hunk 2]
```diff
@@ -43,6 +50,10 @@ def run_command(
         "octoxlabs-get-user-by-username": get_user_by_username,
         "octoxlabs-get-groups": get_groups,
         "octoxlabs-get-permissions": get_permissions,
+        "octoxlabs-search-scroll-devices": search_scroll_devices,
+        "octoxlabs-search-scroll-users-inventory": search_scroll_users,
```
Comment (MLainer1) [inline Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.py:54]: This command is missing from the .yml file

File: Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml [hunk 3]
```diff
@@ -584,12 +595,237 @@ script:
...
+        name: scroll_id
+    outputs:
+      - contextPath: OctoxLabs.ScrolledDevices.count
+        description: 'Total devices count.'
+        type: Number
+      - contextPath: OctoxLabs.ScrolledDevices.scroll_id
+        description: 'Specific Scroll Id'
+        type: String
+      - contextPath: OctoxLabs.ScrolledDevices.results
+        description: 'List<Dict> Device information.'
```
Comment (MLainer1) [inline Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml:730]:
```suggestion
        description: 'List Device information.'
```

File: Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml [hunk 4]
```diff
@@ -584,12 +595,237 @@ script:
...
+        description: 'Query.'
+        name: query
+      - default:
+        description: 'Fields.'
+        isArray: true
+        name: fields
+      - default:
+        description: 'Page.'
+        name: page
+      - default:
```
Comment (MLainer1) [inline Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml:663]: In the code it looks like there is a default size: 50. It is better to add this in the description

File: Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml [hunk 5]
```diff
@@ -584,12 +595,237 @@ script:
...
+    description: 'Search in your AVM'
+    arguments:
+      - default:
+        description: 'Query.'
+        name: query
+      - default:
+        description: 'Fields.'
+        isArray: true
+        name: fields
+      - default:
```
Comment (MLainer1) [inline Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml:660]: In the code it looks like there is a default size: 1. It is better to add this in the description

File: Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml [hunk 6]
```diff
@@ -584,12 +595,237 @@ script:
...
+    description: 'Search in your Applications'
+    arguments:
+      - default:
+        description: 'Query.'
+        name: query
+      - default:
+        description: 'Fields.'
+        isArray: true
+        name: fields
+      - default:
```
Comment (MLainer1) [inline Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml:634]: In the code it looks like there is a default size: 1. It is better to add this in the description

File: Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml [hunk 7]
```diff
@@ -584,12 +595,237 @@ script:
...
+        description: 'Page.'
+        name: page
+      - default:
```
Comment (MLainer1) [inline Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml:637]: In the code it looks like there is a default size: 50. It is better to add this in the description

File: Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.py [hunk 8]
```diff
@@ -28,7 +30,12 @@ def run_command(
         "octoxlabs-get-discoveries": get_discoveries,
         "octoxlabs-get-last-discovery": get_last_discovery,
         "octoxlabs-search-devices": search_devices,
+        "octoxlabs-search-users-inventory": search_users_inventory,
```
Comment (MLainer1) [inline Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.py:33]: This command is missing from the .yml file

File: Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.py [hunk 9]
```diff
@@ -6,6 +6,8 @@
 import urllib3
 from typing import Any, Dict, List, Callable, Optional
 
+# from Packs.Base.Scripts.CommonServerPython.CommonServerPython import CommandResults
+
```
Comment (MLainer1) [inline Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.py:10]: Is it necessary?

File: Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml [hunk 10]
```diff
@@ -15,6 +15,17 @@ configuration:
...
+  display: HTTPS Proxy
+  required: false
+  type: 0
+  additionalinfo: Your HTTPS Proxy URL
+- name: no_verify
+  required: false
+  type: 8
+  display: No Verify
+  additionalinfo: Don't Verify SSL
+
```
Comment (MLainer1) [inline Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.yml:28]: Those 2 new parameters are not documented in the release notes

**Review by MLainer1 — CHANGES_REQUESTED** [review]
Thank you for your contribution.
 A few notes:
 1. Make sure to add a human readable output for every command.
 2. Add tests for the new commands.

Let me know if you need any help/

**Comment by ogulcanhero** [comment]
Hi @MLainer1, thank you for your review. I'm having a problem with the demisto-sdk lint command. When I run the code, I get the error lint() got an unexpected keyword argument 'console_log_threshold'. Is this a known bug?

**Comment by MLainer1** [comment]
> Hi @MLainer1, thank you for your review. I'm having a problem with the demisto-sdk lint command. When I run the code, I get the error lint() got an unexpected keyword argument 'console_log_threshold'. Is this a known bug?

Hi @ogulcanhero, it is a known issue, and a fix will be out in the next few days.
You can use the `pre-commit` command in the mean time.

**Comment by ogulcanhero** [comment]
Hi @MLainer1, I have 2 questions;

- Pre-commit is completely passed in my local, I couldn't understand the reason for the error here.
- Is there anything I need to do for the Legacy Validate test?

Thank You!

**Comment by MLainer1** [comment]
Hi @ogulcanhero, are you running the `demisto-sdk pre-commit` command? Is your demisto-sdk up to date with the latest version?

**Comment by ogulcanhero** [comment]
Hi @MLainer1 ,
Yes, I use `demisto-sdk pre-commit` and I am sure that demisto-sdk is in the latest version because when I had a problem with lint, the first thing I did was to update demisto-sdk.

**Comment by MLainer1** [comment]
@ogulcanhero It looks like the pre-commit fails to add a dot at the end of some sentences. I think that adding that manually shouldn't take more than a few minutes.

**Comment by ogulcanhero** [comment]
Hi @MLainer1, 
I think everything is ok. When can you merge.
Thank You!

**Comment by MLainer1** [comment]
Hi @ogulcanhero, let's schedule a demo to review your changes and make sure everything works as expected. I'll reach out through email to schedule it.

**Comment by ogulcanhero** [comment]
Hi @MLainer1 , 
Thank you for the review. Is it possible for us to make the improvements you mentioned for now in the next version because our customer has been waiting for the features we added for a long time?
Have a nice day!

**Comment by MLainer1** [comment]
@ogulcanhero Yes, as the code works and my suggestions are for the visibility of the commands (human readable and better error management)

**Comment by MLainer1** [comment]
@ogulcanhero, is the PR finished on your side?

**Comment by ogulcanhero** [comment]
Yes @MLainer1 , Thank you!

**Review by MLainer1 — APPROVED** [review]

**Pull request merged**